### PR TITLE
add tensorflow filter

### PR DESCRIPTION
**Pull request by kpango**
Signed-off-by: kpango <kpango@vdaas.org>

### Description:
add ingress filter for tensorflow GenVector.
this PR uses #948's interface, please review this PR after PR #948 merged/

### Environment:

- Go Version: 1.15.7
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.13.1

### Types of changes:

- [ ] Bug fix [type/bug]
- [x] New feature [type/feature]
- [ ] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [ ] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .github/workflows/dockers-filter-ingress-tensorflow.yaml [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+          image-ref: "${{ steps.image_name.outputs.IMAGE_NAME }}:${{ steps.determine_tag.outputs.TAG_NAME }}"
+          format: "table"
+      - name: Run vulnerability scanner (sarif)
+        if: startsWith( github.ref, 'refs/tags/')
+        uses: aquasecurity/trivy-action@master
+        with:
+          image-ref: "${{ steps.image_name.outputs.IMAGE_NAME }}:${{ steps.determine_tag.outputs.TAG_NAME }}"
+          format: "template"
+          template: "@/contrib/sarif.tpl"
+          output: "trivy-results.sarif"
```
Comment (rinx) [inline .github/workflows/dockers-filter-ingress-tensorflow.yaml:163]: Please follow the latest workflow.
https://github.com/vdaas/vald/blob/master/.github/workflows/dockers-agent-ngt-image.yml#L84-L106

File: dockers/filter/ingress/tensorflow/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+WORKDIR ${GOPATH}/src/github.com/${ORG}/${REPO}/Makefile.d
+COPY Makefile.d .
+
+WORKDIR ${GOPATH}/src/github.com/${ORG}/${REPO}
+COPY Makefile .
+COPY .git .
+
+RUN make REPO=${ORG} NAME=${REPO} cmd/${PKG}/${APP_NAME} \
+    && cp "cmd/${PKG}/${APP_NAME}" "/usr/bin/${APP_NAME}" \
+    && make tensorflow/install
```
Comment (rinx) [inline dockers/filter/ingress/tensorflow/Dockerfile:82]: Is this really needed?
I think the execution of `make cmd/${PKG}/${APP_NAME}` will call `tensorflow/install` task.

File: dockers/filter/ingress/tensorflow/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+ARG GO_VERSION=latest
+ARG DISTROLESS_IMAGE=gcr.io/distroless/cc
+ARG DISTROLESS_IMAGE_TAG=nonroot
+ARG UPX_OPTIONS=-9
+ARG MAINTAINER="vdaas.org vald team <vald@vdaas.org>"
+
+FROM golang:${GO_VERSION} AS builder
+
+ARG UPX_OPTIONS
```
Comment (rinx) [inline dockers/filter/ingress/tensorflow/Dockerfile:25]: Did you try to use `upx`? If it doesn't work well, please delete `UPX_OPTIONS` from both of this file and Makefile.d/docker.mk.

File: dockers/filter/ingress/tensorflow/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+FROM ${DISTROLESS_IMAGE}:${DISTROLESS_IMAGE_TAG}
+LABEL maintainer "${MAINTAINER}"
+
+ENV APP_NAME tensorflow
+
+COPY --from=builder /usr/bin/${APP_NAME} /go/bin/${APP_NAME}
+COPY --from=builder /usr/local/lib/libtensorflow* /usr/local/lib/
+
+ENV LD_LIBRARY_PATH=/usr/local/lib
```
Comment (rinx) [inline dockers/filter/ingress/tensorflow/Dockerfile:92]: badge
```suggestion
ENV LD_LIBRARY_PATH /usr/local/lib
```

File: Makefile.d/docker.mk [hunk 5]
```diff
@@ -294,6 +295,22 @@ docker/build/manager-index:
 	    --build-arg MAINTAINER=$(MAINTAINER) \
 	    --build-arg UPX_OPTIONS=$(UPX_OPTIONS)
 
+.PHONY: docker/name/filter-ingress-tensorflow
+docker/name/filter-ingress-tensorflow:
+	@echo "$(ORG)/$(MANAGER_INDEX_IMAGE)"
```
Comment (rinx) [inline Makefile.d/docker.mk:300]:
```suggestion
	@echo "$(ORG)/$(FILTER_INGRESS_TF_IMAGE)"
```

File: Makefile.d/build.mk [hunk 6]
```diff
@@ -472,6 +473,39 @@ cmd/manager/replication/controller/controller: \
 		-o $@ \
 		$(dir $@)main.go
 
+cmd/filter/ingress/tensorflow/tensorflow: \
+	tensorflow/install \
+	$(GO_SOURCES_INTERNAL) \
+	$(PBGOS) \
+	$(shell find ./cmd/agent/core/ngt -type f -name '*.go' -not -name '*_test.go' -not -name 'doc.go') \
+	$(shell find ./pkg/agent/core/ngt ./pkg/agent/internal -type f -name '*.go' -not -name '*_test.go' -not -name 'doc.go')
```
Comment (rinx) [inline Makefile.d/build.mk:481]:
```suggestion
	$(shell find ./cmd/filter/ingress/tensorflow -type f -name '*.go' -not -name '*_test.go' -not -name 'doc.go') \
	$(shell find ./pkg/filter/ingress/tensorflow -type f -name '*.go' -not -name '*_test.go' -not -name 'doc.go')
```

**Review by rinx — CHANGES_REQUESTED** [review]

File: .github/workflows/dockers-filter-ingress-tensorflow.yaml (hunk 1 above)
Comment (rinx) [reply .github/workflows/dockers-filter-ingress-tensorflow.yaml]: @datelier Thanks for your fix. But additionaly, please rewrite the `image-ref` field in vulnerability scanner sections like the following.
https://github.com/vdaas/vald/blob/master/.github/workflows/dockers-agent-ngt-image.yml#L97

**Comment by rinx** [comment]
@datelier Just added one comment.
https://github.com/vdaas/vald/pull/951#discussion_r594826028

File: .github/workflows/dockers-filter-ingress-tensorflow.yaml (hunk 1 above)
Comment (datelier) [reply .github/workflows/dockers-filter-ingress-tensorflow.yaml]: :+1:

**Review by rinx — APPROVED** [review]

**Review by kpango — COMMENTED** [review]
LGTM

**Pull request merged**